operatortrace-go/pkg/client/tracing_interface.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package client
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/trace"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TracingClient extends client.Client with tracing functionality
+type TracingClient interface {
+	client.Client
+	trace.Tracer
```

Comment:
nit: Embedding this also brings the unexported method `tracer()` which you kind of don't need and may interfere with making mocks and extending it's functionality. You really only need  

`Start(ctx context.Context, spanName string, opts ...SpanStartOption) (context.Context, Span)`
because whatever other tracer satisfies that method, etc. 
This is non-blocking.

Reply:
If you were to create an interface 
```go

type Tracer interface {
	Start(ctx context.Context, spanName string, opts ...SpanStartOption) (context.Context, Span)
}
```

You could embed that rather than the library tracer because of the unexported method they embed in their interface.  You can resolve this.